libs/state/src/lib/rx-state.service.ts

```diff
@@ -115,33 +115,30 @@ export class RxState<T extends object> implements OnDestroy, Subscribable<T> {
    * @return T | Partial<T>
    */
 
-  get<K1 extends keyof T>(k1?: K1): T | Partial<T>;
+  get<K1 extends keyof T>(k1?: K1): T[K1];
```

Comment:
souldnt this be  T |  T[K1]? Also if we are abet to pass also no param we may move that in a custom overload?

Reply:
If you take a look in the `safe-pluck.ts` you will see that if you pass 4 keys into the function it is possible to return the value for the 2nd key (if the 3rd key returns `undefined`). So logically yes! it should return `T | T[K1]` but if I keep `T |` TS will return error as below.

this is how I reproduce it:
**rx-state.service.ts:**
![image](https://user-images.githubusercontent.com/24934035/89041272-62ae0f80-d345-11ea-9b74-8ad038e38ec7.png)
**component.ts:**
![image](https://user-images.githubusercontent.com/24934035/89041355-7eb1b100-d345-11ea-8022-682ad86c8be7.png)
error message:
`Type 'SomeClass | { entity: SomeClass; }' is not assignable to type 'SomeClass'.
  Property 'test' is missing in type '{ entity: SomeClass; }' but required in type 'SomeClass'.`

The fix that I provided resolves this error.

Reply:
> Also if we are abet to pass also no param we may move that in a custom overload?

That was what I was thinking. So instead of just `get<K1 extends keyof T>(k1?: K1): T[K1];`, there is:
```ts
get(): T;
get<K1 extends keyof T>(k1: K1): T[K1];
```